### PR TITLE
[Need Help] Refactoring: Put the downloaded binaries in a bin directory and don't rename them to the binaryName

### PR DESCRIPTION
**Pull request by uAliFurkanY**
# The `run` command does not work.
## We need a way to set `NL_PATH` without editing JSON.  See [#372](https://github.com/neutralinojs/neutralinojs/issues/372)

`downloader.js` changes:
- After downloading, create a `bin` directory
- Copy the binary files to the newly-created `bin` directory

`bundler.js` changes: 
- Remove unused import
- Adjust paths to support new ones and legacy ones

`runner.js` changes:
- Adjust paths to support new ones and legacy ones
- Fix 777 chmod which could lead to vulnerabilities
- Remove unusued import

**What works**  
[x] Creating project  
[x] Bundling & running  
[ ] Using the run command (Can't use `--load-dir-res`)
[ ] Using the listen command (Same error as ^)

Error message  
```
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be string, but is null
Aborted (core dumped)
```

### CODE REVIEW

**Comment by shalithasuranga** [comment]
Hey. @uAliFurkanY  Thanks for the PR. Really like the code clean-up and chmod improvements. Regarding paths-related change. I didn't cleary understand the goal. Could you please explain about me. Did you face any problem with `neu run` or is this just to keep binaries in a separate folder?

**Comment by uAliFurkanY** [comment]
It is to keep binaries in a separate folder. Also, it makes it way easier to write your own bundler with this method. Will make future contributions easier as well.

**Comment by shalithasuranga** [comment]
Yeah agree.. I just checked the npm bundle's source of Electron. They also follow a similar approach. We also can use the same way and will be familiar for developers who migrate from Electron. I added this idea to the [roadmap](https://github.com/neutralinojs/proposals). I will implement requirement stuff in the server code and let you know here. Electron uses the first argument to decide the path context. Eg: `electron ./myapp`. I will check and fix the other issue you added. After that, we can merge this one.

**Comment by uAliFurkanY** [comment]
Thanks! Have a nice weekend

**Comment by uAliFurkanY** [comment]
Hello, how is the development going?  
I realized that you can't properly pack apps without this patch.
Like, a proper `.deb` can not be packed that follows standarts.  

Here's how you are supposed to pack a `.deb` for a self-contained app:
Packing  
Here's me installing the `.deb` and running the app:  
Installing, Running

Here's a quick fix for this:  
The neutralino executable should look for `res.neu` in the directory it is, not your working directory.

Here's a quick workaround, use a shell script to `cd` into the app directory:  
Packing  
Here are the `tree`s for consistency:  
image  
And it installs and launches correctly!  
Installing, Running  
This workaround might not be up to spec, but is acceptable in self-contained apps. But we still need the argument support for `neu run` and `neu listen` :P

**Comment by shalithasuranga** [comment]
Hey I will try creating deb package like you mentioned and reproduce the issue. Not sure it's related to [this](https://github.com/neutralinojs/neutralinojs/issues/375) issue. I already added this issue to v2.2.0 [milestone](https://github.com/neutralinojs/neutralinojs/milestone/13). I will add path changing support and let you know :)

**Comment by shalithasuranga** [comment]
@uAliFurkanY  I think it's nice to use Electron's way of packaging deb. I just checked and the following structure and worked without `cd`.

- I added `deb-test-linux` app to `/usr/share/deb-test`
- After that, I added the following shell script to `/usr/bin/deb-test`
```sh
#!/bin/sh
nohup /usr/share/deb-test/deb-test-linux 2>&1 &
```
For reference, I used the Skype Debian package.

But still, your idea is awesome to pass the app path. I will implement it asap.

**Review by shalithasuranga — APPROVED** [review]
Thanks :+1:

**Comment by shalithasuranga** [comment]
Hey.. these changes are merged. `--path=<path>` was added to the binary. I will make a release soon with this feature.

**Comment by shalithasuranga** [comment]
Btw. @uAliFurkanY  I would like to add you to https://neutralino.js.org/docs/community/committers. Okay to add your name as in GitHub?

**Comment by uAliFurkanY** [comment]
> Btw. @uAliFurkanY I would like to add you to https://neutralino.js.org/docs/community/committers. Okay to add your name as in GitHub?

Yes, that would be great.

> I added deb-test-linux app to `/usr/share/deb-test'

I used `/opt`, because it's for self contained apps like Electron apps. `/usr/share` is generally used for architecture-independent files for programs.

**Pull request merged**
